airflow/api_fastapi/core_api/datamodels/ui/config.py

```diff
@@ -0,0 +1,50 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from pydantic import BaseModel
+
+
+class Config(BaseModel):
+    """configuration serializer."""
+
+    navbar_color: str
+    navbar_text_color: str
+    navbar_hover_color: str
+    navbar_text_hover_color: str
+    navbar_logo_text_color: str
+    page_size: int
+    auto_refresh_interval: int
+    default_ui_timezone: str
+    hide_paused_dags_by_default: bool
+    instance_name: str
+    instance_name_has_markup: bool
+    enable_swagger_ui: bool
+    require_confirmation_dag_change: bool
+    default_wrap: bool
+    warn_deployment_exposure: bool
+    audit_view_excluded_events: str
+    audit_view_included_events: str
+    is_k8s: bool
+    test_connection: str
+    state_color_mapping: dict
```

Comment:
Shouldn't this be in the `settings` part of the response ? (IS_K8S_OR_K8SCELERY_EXECUTOR) too ?

(maybe I missed something and we decided to flatten the response, which i'm not against either)

If we do not have `settings` and `config` and everything is under flattened, we could just return a plain:
```json
{
"navbar_color": ...,
"other_key": ...
}
```

WIthout needing the extra level of nesting of `config` inside `config`

Reply:
@bbovenzi suggested to remove `settings` and `config` and return flattened dict.  Agreed @pierrejeambrun removed config nesting.